android/src/main/java/co/boundstate/BranchDeepLinks.java

```diff
@@ -289,6 +293,88 @@ public void onLogoutFinished(boolean loggedOut, BranchError error) {
             );
     }
 
+    @PluginMethod
+    public void getBranchQRCode(final PluginCall call) throws JSONException {
+        JSObject analytics = call.getObject("analytics", new JSObject());
+        JSObject properties = call.getObject("properties", new JSObject());
+        LinkProperties linkProperties = new LinkProperties();
```

Comment:
Capturing our discussion:
Have an overloaded constructor on the native side for LinkProperties which accepts a serializable JSONObject/JSObject and move  the setting logic to that constructor.
LinkProperties linkProperties = new LinkProperties(analytics);